sdk/dotnet/Pulumi.Automation/Exceptions/ProjectSettingsConflictException.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright 2016-2021, Pulumi Corporation
+
+using System;
+
+namespace Pulumi.Automation.Exceptions
+{
+    internal class ProjectSettingsConflictException : Exception
```

Comment:
Does Pulumi have a policy whether exceptions should be public? 

Can consider making exceptions inheriting from `System.Exception` public:
> An internal exception is only visible inside its own internal scope. After the exception falls outside the internal scope, only the base exception can be used to catch the exception. If the internal exception is inherited from Exception, SystemException, or ApplicationException, the external code will not have sufficient information to know what to do with the exception.

Source: https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1064

Reply:
Right. I was stumped about RS0016 I got when trying to make this public but looks like I found what it's about, adding some notes to README.md and a comment on the new exception.

Reply:
I think it should be public with an internal constructor, That way it can be caught explicitly if for any reason a consumer wishes to do so.